src/run-codemod-on-file.ts

```diff
@@ -138,20 +138,21 @@ export default async function runCodemodOnFile(
       parse(source: string, opts: Record<string, unknown>) {
         const babelOpts = {
           ...getBabelOpts(),
-          ..._.pick(codemod, 'presets'),
+          ..._.pick(codemod, 'babelTransformOptions'),
           // There are options that are recognized by recast but not babel. Babel errors when they're passed. To avoid
           // this, we'll omit them.
           ..._.omit(
             opts,
             'jsx', 'loc', 'locations', 'range', 'comment', 'onComment', 'tolerant', 'ecmaVersion'
           ),
-          /**
-           * We must have babel emit tokens. Otherwise, recast will use esprima to tokenize, which won't have the
-           * user-provided babel config.
-           *
-           * https://github.com/benjamn/recast/issues/834
-           */
           parserOpts: {
+            ..._.pick(codemod, 'babelTransformOptions.parserOpts'),
```

Comment:
`pick` uses the second argument as a key – it doesn't do path parsing. You'd want `..._.pick(codemod. babelTransformOptions, 'parserOpts')`. 

I fixed this object spreading on my local branch.